addons/mrp/models/mrp_workorder.py

```diff
@@ -150,27 +151,24 @@ def _read_group_workcenter_id(self, workcenters, domain):
                                      domain="[('allow_workorder_dependencies', '=', True), ('id', '!=', id), ('production_id', '=', production_id)]",
                                      copy=False)
 
-    @api.depends('production_availability', 'blocked_by_workorder_ids.state')
+    @api.depends('production_availability', 'blocked_by_workorder_ids.state', 'blocked_by_workorder_ids.qty_producing', 'qty_ready')
     def _compute_state(self):
         # Force to compute the production_availability right away.
         # It is a trick to force that the state of workorder is computed at the end of the
         # cyclic depends with the mo.state, mo.reservation_state and wo.state and avoid recursion error
         self.mapped('production_availability')
         for workorder in self:
-            if workorder.state == 'pending':
-                if all([wo.state in ('done', 'cancel') for wo in workorder.blocked_by_workorder_ids]):
-                    workorder.state = 'ready' if workorder.production_availability == 'assigned' else 'waiting'
-                    continue
-            if workorder.state not in ('waiting', 'ready'):
-                continue
-            if not all([wo.state in ('done', 'cancel') for wo in workorder.blocked_by_workorder_ids]):
-                workorder.state = 'pending'
+            if workorder.state in ('done', 'cancel'):
                 continue
-            if workorder.production_availability not in ('waiting', 'confirmed', 'assigned'):
+            blocked = workorder._are_previous_workorder_blocking()
+            if workorder.state == 'progress' and not blocked:
                 continue
-            if workorder.production_availability == 'assigned' and workorder.state == 'waiting':
-                workorder.state = 'ready'
-            elif workorder.production_availability != 'assigned' and workorder.state == 'ready':
+            has_qty_ready = workorder._has_qty_ready()
+            if blocked and not has_qty_ready:
+                workorder.state = 'pending'
+            elif workorder.production_availability == 'assigned' or (has_qty_ready and workorder.blocked_by_workorder_ids):
```

Comment:
It's super strange to ignore the BoM configuration in that case (produce once everything is ready or asap)

Reply:
It's been tested by @mathieu-mgm, it's ok since the bom configuration is intended for the first WO and as the more dynamic state transition is then for the following wo